### DISCUSSION TITLE
Call JS Functions from Rust but generated at `create-tauri-app`

### DISCUSSION THREAD

**Discussion by TuxPenguin09** [tauri-apps/tauri#4618] [Q&A]
Same as #3018 and saw the suggested answer and how to do that `window`, got error after I tried said `it doesn't have scope for window.eval`. I will be asking the same but for generated `create-tauri-app`. Is there any way or a code to call JavaScript along with a default code at `main.rs` just for something like "Open New File" and appears a prompt inside the web view. New here at Tauri something, also a mid-beginner at Rust.

main.rs (default code at `create-tauri-app`):
```
fn main() {
  let context = tauri::generate_context!();
  tauri::Builder::default()
  .menu(if cfg!(target_os = "macos") {
    tauri::Menu::os_default(&context.package_info().name)
  } else {
    tauri::Menu::default()
  })
  .run(context)
  .expect("error while running editor");
```

**Comment by simonhyll** [comment]
Just to be clear here, what you want is a "Open new file" item to appear in the menu, and clicking it should pop open a dialog inside the WebView? Because that sounds like something that might be possible to add to the templates for various frameworks that `create-tauri-app` supports, but it wouldn't make much sense unless all frameworks support it by default, which they don't. Preferably the templates shouldn't contain a bunch of sample code, they're meant more for getting started quickly in a framework you're already familiar with, not teach you how to use them.

If you head on into the Discord there's a channel called #taurist where if my understanding of the project is correct it's more of a samples/tutorials kind of project that might show you how to do stuff like that, but it's afaik in very early development.

**Reply by TuxPenguin09** [reply]
I mean yes, that "Open New File" in a menu bar I followed the Tauri docs and it shows in the webview made in HTML and CSS,
like this
<img width="126" alt="image" src="https://user-images.githubusercontent.com/31815005/177906169-7daa1a13-631f-45e3-9e46-5ce8d1482703.png"> 
<img width="764" alt="image" src="https://user-images.githubusercontent.com/31815005/177905831-ee588e8a-21cb-4da7-b964-9405586971ba.png"> 
(just used the console thing in chrome to popup this)

not like this as in the 'os' thing
<img width="655" alt="image" src="https://user-images.githubusercontent.com/31815005/177905235-e6ccadd7-1c7f-4408-a6a5-0914225e3b01.png">
